addons/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/handler/IhcHandler.java

```diff
@@ -0,0 +1,920 @@
+            if (!systemInfo.getDatalineVersion().isEmpty()) {
+                properties.put(PROPERTY_DATALINE_VERSION, systemInfo.getDatalineVersion());
+            }
+            if (!systemInfo.getRfModuleSerialNumber().isEmpty()) {
+                properties.put(PROPERTY_RF_MODULE_SERIALNUMBER, systemInfo.getRfModuleSerialNumber());
+            }
+            if (!systemInfo.getRfModuleSoftwareVersion().isEmpty()) {
+                properties.put(PROPERTY_RF_MODULE_VERSION, systemInfo.getRfModuleSoftwareVersion());
+            }
+            properties.put(PROPERTY_PROJECT_DATE,
+                    projectInfo.getLastmodified().getAsLocalDateTime().atZone(ZoneId.systemDefault()).toString());
+            properties.put(PROPERTY_PROJECT_NUMBER, projectInfo.getProjectNumber());
+            updateProperties(properties);
+        } catch (IhcExecption e) {
+            logger.warn("Controller information fetch failed, reason:  {}", e.getMessage());
+        }
+    }
+
+    private void updateControllerInformationChannels() {
+        try {
+            WSSystemInfo systemInfo = ihc.getSystemInfo();
+            logger.debug("Controller information: {}", systemInfo);
+
+            updateState(new ChannelUID(getThing().getUID(), CHANNEL_CONTROLLER_UPTIME),
+                    new DecimalType((double) systemInfo.getUptime() / 1000));
+        } catch (IhcExecption e) {
+            logger.warn("Controller uptime information fetch failed, reason:  {}", e.getMessage());
+        }
+    }
+
+    private void updateControllerTimeChannels() {
+        try {
+            WSTimeManagerSettings timeSettings = ihc.getTimeSettings();
+            logger.debug("Controller time settings: {}", timeSettings);
+
+            ZonedDateTime time = timeSettings.getTimeAndDateInUTC().getAsZonedDateTime(ZoneId.of("Z"))
+                    .withZoneSameInstant(ZoneId.systemDefault());
+            updateState(new ChannelUID(getThing().getUID(), CHANNEL_CONTROLLER_TIME), new DateTimeType(time));
+        } catch (IhcExecption e) {
+            logger.warn("Controller uptime information fetch failed, reason:  {}", e.getMessage());
+        }
+    }
+
+    private void updateResourceChannel(ChannelUID channelUID, Command command) {
+        try {
+            Channel channel = thing.getChannel(channelUID.getId());
+            if (channel != null) {
+                ChannelParams params = new ChannelParams(channel);
+                logger.debug("Channel params: {}", params);
+                if (params.isDirectionReadOnly()) {
+                    logger.debug("Read only channel, skip the update to {}", channelUID);
+                    return;
+                }
+                updateChannel(channelUID, params, command);
+            }
+        } catch (IhcExecption e) {
+            logger.error("Can't update channel '{}' value, cause ", channelUID, e.getMessage());
+        } catch (IllegalArgumentException e) {
```

Comment:
If this exception catching is magic compared with the other code don't forget to update it :-)

Reply:
Done